client/my-sites/email/add-mailboxes/index.tsx

```diff
@@ -107,26 +107,55 @@ const useAdditionalProps = ( {
 const recordClickEvent = ( {
 	eventName,
 	eventProps = {},
+	provider,
 	selectedDomainName,
 	source = '',
 }: {
 	eventName: string;
 	eventProps?: Record< string, unknown >;
+	provider: EmailProvider;
 	selectedDomainName: string;
 	source?: string;
 } ) => {
 	recordTracksEvent( eventName, {
 		...eventProps,
 		domain_name: selectedDomainName,
-		provider: TITAN_PROVIDER_NAME,
+		provider: isTitan( provider ) ? TITAN_PROVIDER_NAME : GOOGLE_PROVIDER_NAME,
 		source,
 	} );
 };
 
+const WithVerificationGate = ( {
+	children,
+	productFamily,
+	provider,
+}: {
+	children: JSX.Element;
+	productFamily: string;
+	provider: EmailProvider;
+} ): JSX.Element => {
```

Comment:
```suggestion
}: React.PropsWithChildren< {
	productFamily: string;
	provider: EmailProvider;
} > ) => {
```

React offers this pretty neat type for automatically handling different types of `children` props – `React.PropsWithChildren< Props >`. I reacted to the fact that you were using a fragment on line 352 and I'm guessing that's maybe because Typescript complains if you are passing multiple child elements? With this approach, the fragment shouldn't be necessary (unless it's there for some other reason).

I'm also wondering: Typescript infers the return type here, right? Do we gain something from explicitly setting it?

Reply:
Ha, I see what you are talking about, but I'm really used to specifying the children type manually. There are schools of thought that prefer the declarative approach, but after seeing that it's heavily used in Calypso, I don't mind using that approach.

I'm also used to using the fragment out of habit mostly, although in some other cases I have specified the children type as `ReactNode | ReactNode[] | undefined`.